alune/adb.py

```diff
@@ -65,7 +67,33 @@ async def _load_rsa_signer(self):
 
         self._rsa_signer = PythonRSASigner(pub=public_key, priv=private_key)
 
-    async def _connect_to_device(self, port: int):
+    async def scan_localhost_devices(self) -> int | None:
+        """
+        Try to connect with a fast abort on every open localhost port.
+
+        Returns:
+            The first valid open ADB port or None if there wasn't one.
+        """
+        logger.info("Scanning local ports for an open ADB connection...")
+
+        connections = [
+            conn for conn in psutil.net_connections("tcp4") if conn.laddr.port >= 5555 and conn.status == "LISTEN"
```

Comment:
Do we think there's a logical max we should limit it to, or just say screw it and go to the max available? 😅

Reply:
In practice we need to check all ports, but I'll add a warning that it may take some time on more than 10 open ports